File: practice/algorithms/implementation/designer_pdf_viewer/designer_pdf_viewer.hs
module Main where
import Data.Char

getHeight :: String -> [Int] -> Int
getHeight w h = getHeight' w h 0
    where
        getHeight' []    h max_h = max_h
        getHeight' (c:w) h max_h = getHeight' w h (max max_h curr_h)
            where
                pos = ord c - (ord 'a')
                curr_h = h !! pos

main :: IO ()
main = do
    h_temp <- getLine
    word <- getLine
    let h = map read $ words h_temp :: [Int]
        max_h = getHeight word h
        r = length word * max_h
    print r

File: practice/algorithms/implementation/designer_pdf_viewer/designer_pdf_viewer.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    vector<int> h(26);
    for (int h_i = 0; h_i < 26; ++h_i)
    {
       cin >> h[h_i];
    }
    
    string word;
    cin >> word;
    
    int max_h = 0;
    for (int pos = 0; pos < word.size(); ++pos)
    {
        int curr_pos = word[pos] - 'a';
        max_h = max(max_h, h[curr_pos]);
    }
    
    cout << max_h * word.size() << endl;
    
    return 0;
}

